Floor.java

```
@@ -33,46 +25,70 @@ public Floor(Scheduler scheduler) {
     }
 
-    public void setRequestUpButton(Boolean requestUpButton){
+    public void setRequestUpButton(Boolean requestUpButton) {
         this.requestUpButton = requestUpButton;
     }
 
-    public Boolean getRequestDownButton(){
+    public Boolean getRequestDownButton() {
         return requestDownButton;
     }
 
-    public void setRequestDownButton(Boolean requestDownButton){
+    public void setRequestDownButton(Boolean requestDownButton) {
         this.requestDownButton = requestDownButton;
     }
 
     public String getDirectionLamp(){
         return directionLamp;
     }
 
-    public void setDirectionLamp(String directionLamp){ 
+    public void setDirectionLamp(String directionLamp) { 
         this.directionLamp = directionLamp; 
     }
 
-    public Boolean getRequestUpButtonLamp(){
+    public Boolean getRequestUpButtonLamp() {
         return requestUpButtonLamp;
     }
 
-    public void setRequestUpButtonLamp(Boolean isOn){
+    public void setRequestUpButtonLamp(Boolean isOn) {
         requestUpButtonLamp = isOn;
     }
 
-    public Boolean getRequestDownButtonLamp(){
+    public Boolean getRequestDownButtonLamp() {
         return requestDownButtonLamp;
     }
 
-    public void setRequestDownButtonLamp(Boolean isOn){
+    public void setRequestDownButtonLamp(Boolean isOn) {
         this.requestDownButtonLamp = isOn;
     }
 
+    /**
+     * Handles invalid input requests when parsing the data in data.txt
+     * @param currentFloor the floor being requested from in the data.txt file
+     * @param floorRequest the floor destination in the data.txt file
+     * @param direction the direction the elevator is going
+     * @return true if any of these inputs are invalid (negative, greater than 7 or not "up" or "down")
+     * @return false otherwise
+     */
+    private boolean handleInputErrors(int currentFloor, int floorRequest, String direction) {
+        if (currentFloor < 0 || currentFloor > 7) {
+            return true;
```

maybe throw an exception here with the specific message as to what is wrong in the data.txt file instead of returning true/false